### ISSUE TITLE
Compatibility with Ducks' Insane Skills

### ISSUE THREAD

**Issue by FlyingRatBull** [patrickdevarney/RimworldMortarAccuracy#7]
Thanks for the great mod!

I just seem to have found an incompatibility with Ducks' Insane Skills (Skill levels >20). If the shooting pawn has a skill level higher than 20, the accuracy gets really bad really fast. Pawns with a skill level below 20 are just fine.

The resulting value for calculating the radius seems to go into the negative (the drawn circle seems pretty small, the resulting shot is all over the place). My guess would be [this line](https://github.com/patrickdevarney/RimworldMortarAccuracy/blob/f5f3c98db238a12423d6df492b8db69f0278dbe8/Source/PatsMortarAccuracy/Verb_LaunchProjectile_MortarMod.cs#L63).

**Comment by damianb** [comment]
a good fix would probably be to clamp the skill level to 20 as a max. i'll take a shot at an MR.

**Comment by patrickdevarney** [comment]
Good catch. Insane Skills might not be modifying the SkillRecord.MaxLevel as I had expected.

**Comment by damianb** [comment]
> Good catch. Insane Skills might not be modifying the SkillRecord.MaxLevel as I had expected.

Definitely doesn't appear to.  I see it hooking on `SkillRecord.XpRequiredToLevelUpFrom`, `SkillRecord.Interval`, `SkillRecord.Learn`, `SkillRecord.Level`, and `SkillRecord.LearnRateFactor`, but I do not see any modifications to the MaxLevel at all. It only ever references its own internal MaxLevel from its settings.

**Comment by patrickdevarney** [comment]
Ah, it makes sense why Insane Skills isn't modifying `SkillRecord.MaxLevel`. They still want all the vanilla logic to be based on 0-20 and merely have insane multipliers. They specifically do not want to modify MaxLevel.

In this case, I think clamping the multiplier to be 0->1 should do the trick.

Change this:
```
skillMultiplier = 1 - ((averageSkill - SkillRecord.MinLevel) * (Settings.maxSkillSpreadReduction - Settings.minSkillSpreadReduction) / (SkillRecord.MaxLevel - SkillRecord.MinLevel) + Settings.minSkillSpreadReduction);
```

To this:
```
skillMultiplier = 1 - (Math.Clamp((averageSkill - SkillRecord.MinLevel)  / (SkillRecord.MaxLevel - SkillRecord.MinLevel), 0f, 1f) * (Settings.maxSkillSpreadReduction - Settings.minSkillSpreadReduction) + Settings.minSkillSpreadReduction);
```

I'll commit and test this on the weekend.

**Comment by patrickdevarney** [comment]
Fixed in release 1.3.6 https://github.com/patrickdevarney/RimworldMortarAccuracy/releases/tag/1.3.6
